### ISSUE TITLE
Reverse Added Contacts not working in 3.1.1

### ISSUE THREAD

**Issue by GoogleCodeExporter** [msndevs/msnp-sharp#254]
```
What steps will reproduce the problem?

There seem to be quite a few instances where this fails, for example:

1. Start with two accounts which are already contacts, using WLM remove from 
both accounts the other contact from the contact list, hotmail contacts, and 
allow list.
2. Start one of the accounts in MSNPSharp and the other in WLM
3. Add the account logged into MSNPSharp to the account logged into WLM, more 
often than not ReverseAdded will not fire.

This problem also seems to happen almost every time when added from a 
non-hotmail, but passport authenticated account (eg a gmail account which is 
signed into WLM).

Occasionally some of these problems can be resolved by the account on MSNPSharp 
signing out and then in again, but this is not ideal.

What is the expected output? What do you see instead?

I would expect ReverseAdded to fire every time the MSNPSharp's account has been 
added to someone elses reverse list, regardless of whether they have ever been 
added to the list before and then remove, or whether or not the account is a 
non-hotmail, but passport user.

What version of the product are you using? (MSNPSharp, OS, Mono etc.)

MSNPSharp compiled under Visual Studio 2008, running in Windows 7.

Is your code check out from SVN or download from our download site?

Downloaded from the download site.

Please provide any additional information below:

```

Original issue reported on code.google.com by `jimca...@gmail.com` on 2 Feb 2011 at 10:35

**Comment by GoogleCodeExporter** [comment]
```
This issue was updated by revision r1986.
```

Original comment by `hepha...@gmail.com` on 3 Feb 2011 at 8:48
- Changed state: **WontFix**

**Comment by GoogleCodeExporter** [comment]
```
ReverseAdded is fired only in this situtations:

if (contact.OnPendingList ||
   (contact.OnReverseList && !contact.OnAllowedList && !contact.OnBlockedList))
   {
     NSMessageHandler.ContactService.OnReverseAdded(new ContactEventArgs(contact));
   }

Why?

If a contact blocked by you: You say: I don't want to receive any warning.
If a contact allowed by you: You say: This contact can contact with me.
You can use contact.OnReverseList property when logged in.

If you want to inform any contact changes, use ContactAdded and ContactRemoved 
events. See r1986.

```

Original comment by `hepha...@gmail.com` on 3 Feb 2011 at 8:56

**Comment by GoogleCodeExporter** [comment]
```
Please check out the MSNPSHARP_31_STABLE code base in the svn, your issue has 
already been fixed but not publish as a zip file on download page.
```

Original comment by `freezing...@gmail.com` on 4 Feb 2011 at 3:13

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Thanks, I checked out the SVN after Hepha's responses. Unfortunately the 
ContactAdded event doesn't always seem to fire either. In the new DotMSNClient 
on connection to the server sometimes ContactAdded fires for all users, 
sometimes only a few, and sometimes none.

I'm using this library for a bot, so it is not completely necessary that the 
users are added immediately, for now I am just checking if the user is on the 
forward list when a message is received, if not the user is added.
```

Original comment by `jimca...@gmail.com` on 4 Feb 2011 at 3:17

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Thanks, I checked out the SVN after Hepha's responses. Unfortunately the 
ContactAdded event doesn't always seem to fire either. In the new DotMSNClient 
on connection to the server sometimes ContactAdded fires for all users, 
sometimes only a few, and sometimes none.

I'm using this library for a bot, so it is not completely necessary that the 
users are added immediately, for now I am just checking if the user is on the 
forward list when a message is received, if not the user is added.
```

Original comment by `jimca...@gmail.com` on 4 Feb 2011 at 3:18
